qubesmanager/qube_manager.py

```diff
@@ -61,140 +70,363 @@ def mousePressEvent(self, e):  # pylint: disable=invalid-name
-            vm, "backup_timestamp")
-        table.setItem(row_no, VmManagerWindow.columns_indices[
-            'Last backup'], self.last_backup_widget)
-
-        self.dvm_template_widget = table_widgets.VMPropertyItem(
-            vm, "default_dispvm", check_default=True)
-        table.setItem(row_no, VmManagerWindow.columns_indices['Default DispVM'],
-                      self.dvm_template_widget)
-
-        self.is_dispvm_template_widget = table_widgets.VMPropertyItem(
-            vm, "template_for_dispvms", empty_function=(lambda x: not x))
-        table.setItem(
-            row_no, VmManagerWindow.columns_indices['Is DVM Template'],
-            self.is_dispvm_template_widget)
-
-        self.virt_mode_widget = table_widgets.VMPropertyItem(vm, 'virt_mode')
-        table.setItem(row_no, VmManagerWindow.columns_indices[
-            'Virtualization Mode'], self.virt_mode_widget)
-
-        self.table = table
+        self.qid = vm.qid
+        self.name = self.vm.name
+        self.label = self.vm.label
+        self.klass = self.vm.klass
+        self.state = {'power': "", 'outdated': ""}
+        self.updateable = getattr(vm, 'updateable', False)
+        self.update(True)
 
     def update(self, update_size_on_disk=False, event=None):
         """
-        Update info in a single VM row
-        :param update_size_on_disk: should disk utilization be updated? the
-        widget will extract the data from VM object
+        Update VmInfo
+        :param update_size_on_disk: should disk utilization be updated?
         :param event: name of the event that caused the update, to avoid
         updating unnecessary properties; if event is none, update everything
         :return: None
         """
         try:
-            self.info_widget.update_vm_state()
+            self.state['power'] = self.vm.get_power_state()
+
+            if self.vm.is_running():
+                if hasattr(self.vm, 'template') and \
+                        self.vm.template.is_running():
+                    self.state['outdated'] = "to-be-outdated"
+                else:
+                    for vol in self.vm.volumes.values():
+                        if vol.is_outdated():
+                            self.state['outdated'] = "outdated"
+                            break
+            else:
+                self.state['outdated'] = ""
+
+            if self.vm.klass in {'TemplateVM', 'StandaloneVM'} and \
+                    self.vm.features.get('updates-available', False):
+                self.state['outdated'] = 'update'
+
             if not event or event.endswith(':label'):
-                self.label_widget.update()
+                self.label = self.vm.label
             if not event or event.endswith(':template'):
-                self.template_widget.update()
+                try:
+                    self.template = self.vm.template.name
+                except exc.QubesNoSuchPropertyError:
```

Comment:
This should be AttributeError (exc.QubesNoSuchPropertyError inherits from it, so it will be matched too)